### PR TITLE
@xtina-starr => Add palette, TextUrlInput updates

### PR DESCRIPTION
**Pull request by eessex**
Related to https://artsyproduct.atlassian.net/browse/GROW-564

- Adds @artsy/palette package
- Updates `TextUrlInput` to use `styled-components` and some design system pieces
- Adds key commands for Tab/Esc to close editor without mouseclicks

### CODE REVIEW

File: src/client/apps/settings/client/curations/display/test/components/canvas/canvas_images.test.js [hunk 1]
```diff
@@ -165,22 +176,27 @@ describe('Canvas Images', () => {
...
+        props.campaign.canvas.assets = [
+          {url: 'http://artsy.net/image1.jpg'},
+          {url: 'http://artsy.net/image1.jpg'}
+        ]
         const component = mount(
           <CanvasImages {...props} />
         )
-        component.find('.RemoveButton').at(1).simulate('click')
-        const onChangeArgs = props.onChange.mock.calls[4]
+        component.find('.RemoveButton').at(2).simulate('click')
```
Comment (xtina-starr) [inline src/client/apps/settings/client/curations/display/test/components/canvas/canvas_images.test.js]: I'm curious why you passed the class name versus the actual component similar to how you did in the header test file?

File: src/client/components/rich_text/components/input_url.jsx [hunk 2]
```diff
@@ -18,71 +19,160 @@ export class TextInputUrl extends Component {
...
+export const BackgroundOverlay = styled.div`
+  position: fixed;
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+  z-index: 7;
+`
+
+export const Button = Sans.extend`
```
Comment (xtina-starr) [inline src/client/components/rich_text/components/input_url.jsx]: 🎉

File: src/client/apps/settings/client/curations/display/test/components/canvas/canvas_images.test.js (hunk 1 above)
Comment (eessex) [reply src/client/apps/settings/client/curations/display/test/components/canvas/canvas_images.test.js]: For some reason the click wasn't working here in the same way, can try with `RemoveButtonContainer`.

Comment (xtina-starr) [reply src/client/apps/settings/client/curations/display/test/components/canvas/canvas_images.test.js]: Ah gotcha. Yeah I don't think it's reason to hold up the PR but I know we are trying to transition to passing the actual component itself so I wanted to mention it. 

Maybe see if you can get it to work with `RemoveButtonContainer` and if not no biggy. Feel free to merge when you are ready.

**Comment by xtina-starr** [comment]
It's dope to see @artsy/palette make it's way to positron 💯

File: src/client/apps/settings/client/curations/display/test/components/canvas/canvas_images.test.js [hunk 3]
```diff
@@ -186,17 +186,16 @@ describe('Canvas Images', () => {
...
         props.campaign.canvas.logo = 'http://artsy.net/logo.jpg'
         props.campaign.canvas.assets = [
-          {url: 'http://artsy.net/image1.jpg'},
           {url: 'http://artsy.net/image1.jpg'}
         ]
         const component = mount(
           <CanvasImages {...props} />
         )
-        component.find('.RemoveButton').at(2).simulate('click')
+        component.find(RemoveButtonContainer).at(1).simulate('click')
```
Comment (xtina-starr) [inline src/client/apps/settings/client/curations/display/test/components/canvas/canvas_images.test.js]: 👍

**Pull request merged**